runtime/lua/vim/lsp/util.lua

```diff
@@ -547,13 +547,55 @@ function M.focusable_preview(unique_name, fn)
   end)
 end
 
--- Convert markdown into syntax highlighted regions by stripping the code
--- blocks and converting them into highlighted code.
--- This will by default insert a blank line separator after those code block
--- regions to improve readability.
+--- Trim empty lines from input and pad left and right with spaces
+--- Defaults to left and right padding by 1 for backward compatibility
+--- 
+--@param contents table of lines to trim and pad
+--@param opts dictionary with optional fields
+--             - pad_left  amount of columns to pad contents at left
+--             - pad_right amount of columns to pad contents at right
+--@return contents table of trimmed and padded lines
+function M._trim_and_pad(contents,opts)
+  validate {
+    contents = { contents, 't' };
+    opts = { opts, 't', true };
+  }
+  opts = opts or {}
+  -- Default padding of 1 for backwards compatibility
```

Comment:
@tjdevries I don't think we need the code for backward compatibility because it's still before v0.5.0 release, what do you think?

Reply:
Having compared the two, I now actually prefer (for my use cases) the padding that @norcalli introduced -- so it's less a matter of compatibility than of reasonable defaults (and not needlessly moving cheese). But I could either set this to 0 (which means most people will probably want to override it) or just remove the "for backwards compatibility"?

Reply:
We should leave padding of 1 because it's pretty, but overridable. 

We can remove comment about backwards compatibility though.